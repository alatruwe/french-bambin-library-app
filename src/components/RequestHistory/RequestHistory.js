import React from "react";
import Request from "./Request/Request";
import requestData from "../../requestData.js";

export default class RequestHistory extends React.Component {
  render() {
    return (
      <section>
        <ul>
          {requestData.map((request) => (
            <li key={request.id}>
              <Request
                title={request.title}
                object={request.object}
                message={request.message}
                date={request.date}
              />
            </li>
          ))}
        </ul>
      </section>
    );
  }
}
