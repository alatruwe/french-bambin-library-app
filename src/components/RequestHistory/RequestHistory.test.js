import React from "react";
import ReactDOM from "react-dom";
import RequestHistory from "./RequestHistory";
import renderer from "react-test-renderer";

describe(`RequestHistory component`, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<RequestHistory />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<RequestHistory />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
