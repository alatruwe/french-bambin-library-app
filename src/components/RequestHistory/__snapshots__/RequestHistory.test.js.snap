// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestHistory component renders the UI as expected 1`] = `
<section>
  <ul>
    <li>
      <section>
        <h2>
          Item: 
          Sami et Julie à la plage
        </h2>
        <h1>
          Request object: 
          Hello
        </h1>
        <p>
          Request message: 
          How are you?
        </p>
        <p>
          Sent: 
          Feb 17th 2021
        </p>
      </section>
    </li>
    <li>
      <section>
        <h2>
          Item: 
          Le petit Chaperon rouge
        </h2>
        <h1>
          Request object: 
          Hi!
        </h1>
        <p>
          Request message: 
          I want to get that book please
        </p>
        <p>
          Sent: 
          Feb 17th 2021
        </p>
      </section>
    </li>
    <li>
      <section>
        <h2>
          Item: 
          Le grand imagier de la nature
        </h2>
        <h1>
          Request object: 
          https:/
        </h1>
        <p>
          Request message: 
          Munis de leur seau 
        </p>
        <p>
          Sent: 
          Feb 17th 2021
        </p>
      </section>
    </li>
    <li>
      <section>
        <h2>
          Item: 
          Ludanimo
        </h2>
        <h1>
          Request object: 
          https:/
        </h1>
        <p>
          Request message: 
          Munis de leur seau 
        </p>
        <p>
          Sent: 
          Feb 17th 2021
        </p>
      </section>
    </li>
    <li>
      <section>
        <h2>
          Item: 
          Mon amour
        </h2>
        <h1>
          Request object: 
          https:/
        </h1>
        <p>
          Request message: 
          Munis de leur seau 
        </p>
        <p>
          Sent: 
          Feb 17th 2021
        </p>
      </section>
    </li>
    <li>
      <section>
        <h2>
          Item: 
          Le Verger, jeu de cartes
        </h2>
        <h1>
          Request object: 
          https:/
        </h1>
        <p>
          Request message: 
          Munis de leur seau 
        </p>
        <p>
          Sent: 
          Feb 17th 2021
        </p>
      </section>
    </li>
    <li>
      <section>
        <h2>
          Item: 
          La Fontaine raconte aux enfants
        </h2>
        <h1>
          Request object: 
          https:/
        </h1>
        <p>
          Request message: 
          Munis de leur seau 
        </p>
        <p>
          Sent: 
          Feb 17th 2021
        </p>
      </section>
    </li>
    <li>
      <section>
        <h2>
          Item: 
          La tour Eiffel se balade à Paris
        </h2>
        <h1>
          Request object: 
          https:/
        </h1>
        <p>
          Request message: 
          Munis de leur seau 
        </p>
        <p>
          Sent: 
          Feb 17th 2021
        </p>
      </section>
    </li>
  </ul>
</section>
`;
